config/config.go

```diff
@@ -160,9 +167,20 @@ func (c *S3Cli) allowMultipart() bool {
 			return false
 		}
 	}
+	if c.isAlicloudHost() {
+		return false
+	}
 	return true
 }
 
 func (c *S3Cli) getRegionFromHost() string {
 	return AWSHostToRegion[c.Host]
 }
+
+func (c *S3Cli) isAlicloudHost() bool {
+	return strings.HasPrefix(c.Host, "oss-") && strings.HasSuffix(c.Host, "aliyuncs.com")
+}
+
+func (c *S3Cli) getAlicloudRegionFromHost() string {
+	return strings.TrimSuffix(strings.TrimPrefix(strings.TrimSuffix(c.Host, ".aliyuncs.com"), "oss-"), "-internal")
```

Comment:
Feels like this could maybe be better serviced with a regex? Something like `^oss-(?P<region>.*)(-internal)?.aliyuncs.com$` then get the `region` capture group.

Reply:
We also saw [here](https://www.alibabacloud.com/help/faq-detail/39607.htm) that there may be other segments before `oss-`, how does that affect this code?

Reply:
In the alicloud, this host is oss endpoint and actually the oss sdk and test result can justify it.